examples/webgpu_compute_water.html

```diff
@@ -0,0 +1,511 @@
+					const prevHeight = prevHeightStorage.element( instanceIndex ).toVar();
+
+					const { north, south, east, west } = getNeighborValuesTSL( instanceIndex, heightStorage );
+
+					const neighborHeight = north.add( south ).add( east ).add( west );
+					neighborHeight.mulAssign( 0.5 );
+					neighborHeight.subAssign( prevHeight );
+
+					const newHeight = neighborHeight.mul( viscosity );
+
+					// Get 2-D compute coordinate from one-dimensional instanceIndex.
+					const x = float( instanceIndex.modInt( WIDTH ) ).mul( 1 / WIDTH );
+					const y = float( instanceIndex.div( WIDTH ) ).mul( 1 / WIDTH );
+
+					// Mouse influence
+					const centerVec = vec2( 0.5 );
+					// Get length of position in range [ -BOUNDS / 2, BOUNDS / 2 ], offset by mousePos, then scale.
+					const mousePhase = clamp( length( ( vec2( x, y ).sub( centerVec ) ).mul( BOUNDS ).sub( mousePos ) ).mul( Math.PI ).div( mouseSize ), 0.0, Math.PI );
+
+					newHeight.addAssign( cos( mousePhase ).add( 1.0 ).mul( 0.28 ) );
+
+					prevHeightStorage.element( instanceIndex ).assign( height );
+					heightStorage.element( instanceIndex ).assign( newHeight );
+
+				} )().compute( WIDTH * WIDTH );
+
+				computeSmooth = Fn( () => {
+
+					const height = heightStorage.element( instanceIndex ).toVar();
+					const prevHeight = prevHeightStorage.element( instanceIndex ).toVar();
+
+					// Get neighboring height values.
+					const { north: northH, south: southH, east: eastH, west: westH } = getNeighborValuesTSL( instanceIndex, heightStorage );
+			
+					// Get neighboring prev height values.
+					const { north: northP, south: southP, east: eastP, west: westP } = getNeighborValuesTSL( instanceIndex, prevHeightStorage );
+
+					height.addAssign( northH.add( southH ).add( eastH ).add( westH ) );
+					prevHeight.addAssign( northP.add( southP ).add( eastP ).add( westP ) );
+
+					heightStorage.element( instanceIndex ).assign( height.div( 5 ) );
+					prevHeightStorage.element( instanceIndex ).assign( height.div( 5 ) );
+
+				} )().compute( WIDTH * WIDTH, [ 8, 8 ] );
+
+				// Water Geometry corresponds with buffered compute grid.
+				const waterGeometry = new THREE.PlaneGeometry( BOUNDS, BOUNDS, WIDTH - 1, WIDTH - 1 );
+				// material: make a THREE.ShaderMaterial clone of THREE.MeshPhongMaterial, with customized position shader.
+				const waterMaterial = new THREE.MeshPhongNodeMaterial();
+
+				waterMaterial.lights = true;
+				waterMaterial.colorNode = color( 0x0040C0 );
+				waterMaterial.specularNode = color( 0x111111 );
+				waterMaterial.shininess = Math.max( 50, 1e-4 );
+				waterMaterial.positionNode = Fn( () => {
+
+					// To correct the lighting as our mesh undulates, we have to reassign the normals in the position shader.
+					const { normalX, normalY } = getNormalsFromHeightTSL( vertexIndex, heightRead );
+
+					varyingProperty( 'vec3', 'v_normalView' ).assign( modelNormalMatrix.mul( vec3( normalX, negate( normalY ), 1.0 ) ) );
```

Comment:
TSL now have a function for this: `transformNormalToView( vec3( normalX, negate( normalY ), 1.0 ) )`

Reply:
This seems to break the lighting when I test it. Aren't modelNormalMatrix and modelNormalViewMatrix distinct? And if they aren't, should the name of the node be updated to reflect that?